tests/system/action/poll/test_vote.py

```diff
@@ -577,3 +596,1207 @@ def test_default_vote_weight(self) -> None:
+        )
+        self.update_model("poll/1", {"global_option_id": 11})
+        response = self.request("poll.vote", {"value": "N", "id": 1, "user_id": 1})
+        self.assert_status_code(response, 200)
+        option = self.get_model("option/11")
+        self.assertEqual(option.get("yes"), "0.000000")
+        self.assertEqual(option.get("no"), "1.000000")
+        self.assertEqual(option.get("abstain"), "0.000000")
+
+    def test_global_no_forbidden(self) -> None:
+        self.update_model("poll/1", dict(global_no=False))
+        self.start_poll()
+        response = self.request("poll.vote", {"value": "N", "id": 1, "user_id": 1})
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_global_abstain(self) -> None:
+        self.start_poll()
+        self.create_model(
+            "option/11", {"meeting_id": 113, "used_as_global_option_in_poll_id": 1}
+        )
+        self.update_model("poll/1", {"global_option_id": 11})
+
+        response = self.request("poll.vote", {"value": "A", "id": 1, "user_id": 1})
+        self.assert_status_code(response, 200)
+        option = self.get_model("option/11")
+        self.assertEqual(option.get("yes"), "0.000000")
+        self.assertEqual(option.get("no"), "0.000000")
+        self.assertEqual(option.get("abstain"), "1.000000")
+
+    def test_global_abstain_forbidden(self) -> None:
+        self.update_model("poll/1", dict(global_abstain=False))
+        self.start_poll()
+        response = self.request("poll.vote", {"value": "A", "id": 1, "user_id": 1})
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_negative_vote(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": -1}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_too_many_options(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": 1, "2": 1, "3": 1}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_wrong_options(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"3": 1}},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    @pytest.mark.skip()
+    def test_no_permissions(self) -> None:
+        self.start_poll()
+        # self.make_admin_delegate()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": 1}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_anonymous(self) -> None:
```

Comment:
Same as above: add a test, where the request user is the anonymous.

Reply:
Same as above, not in this PR.